src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -133,6 +135,10 @@ func (r *ClusterReconciler) Reconcile(
 	if proxyAPIExternal != nil && proxyAPIInternal != nil {
 		clusterPorts = append(clusterPorts, resources.NamedServicePort{Name: resources.PandaproxyPortExternalName, Port: proxyAPIInternal.Port + 1})
 	}
+	if redpandaCluster.Spec.Configuration.SchemaRegistry != nil {
```

Comment:
I did not see this code for some time and now I am very confused what is nodePorts, headlessPorts and clusterPorts, I think I understand the first two, but the third one? 🤔

Reply:
NodePorts for external connectivity (Kafka API, Admin API)

HeadlessPorts for internal k8s communication (Admin API, Kafka API and RPC API)

ClusterPorts for schema-registry and panda proxy inside k8s networking boundary. 

The clusterPorts and headlessPorts should be view as services that are used only inside k8s cluster. The key difference is that ports under cluster service can be load balanced thanks to iptables rules. The headless DNS name will be resolved to all POD IPs. Which gives ability to reach to any of the PODs. There is a way to resolve POD ip using headless service.